src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -215,8 +215,8 @@ internal GeneratorDriverState RunGeneratorsCore(Compilation compilation, Diagnos
                     }
 
                     generatorState = ex is null
-                                     ? new GeneratorState(generatorState.Info, postInitSources, inputNodes, outputNodes)
-                                     : SetGeneratorException(MessageProvider, generatorState, sourceGenerator, ex, diagnosticsBag, isInit: true);
+                                     ? new GeneratorState(postInitSources, inputNodes, outputNodes)
+                                     : SetGeneratorException(MessageProvider, GeneratorState.Empty, sourceGenerator, ex, diagnosticsBag, isInit: true);
```

Comment:
>GeneratorState.Empty [](http://example.com/codeflow?start=78&length=20)

It's not clear why this changed. Is it because `generatorState` is equivalent to `GeneratorState.Empty` when `!Initialized`?

Reply:
Yes. If the initialization fails there is no state, and we don't want to try and do anything with it in the future.